src/pages/Group/GroupSettings.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { useMutation, useQueryClient } from "react-query";
+import ToastMessage from "../../components/ToastMessage";
+import Column from "../../components/layout/Column";
+import Row from "../../components/layout/Row";
+import { QueryKeys } from "../../constants";
+import { useApi } from "../../contexts/ApiProvider";
+import { Group } from "../../data/models/Group";
+
+import { useNavigate } from "react-router-dom";
+import LeftArrowIcon from "../../assets/arrow_left.svg";
+import Button from "../../components/Button";
+import Spinner from "../../components/Spinner";
+import TextField from "../../components/TextField";
+import ClosableModal from "../../components/modals/ClosableModal";
+import { useUser } from "../../contexts/UserProvider";
+
+export default function GroupSettings({
+  group,
+  setShowGroupSettings,
+}: {
+  group: Group;
+  setShowGroupSettings: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
+  const { user } = useUser();
+  const navigate = useNavigate();
+
+  // group settings:
+  const [groupNameText, setGroupNameText] = useState("");
+  const [leaveError, setLeaveError] = useState("");
+  const [changeGroupNameError, setChangeGroupNameError] = useState("");
+  const { api } = useApi();
+  const queryClient = useQueryClient();
+
+  // leave modal
+  const confirmLeaveId = "confirm-leave";
+
+  const { mutate: saveGroupName, isLoading: putGroupNameLoading } =
+    useMutation<Group>({
+      mutationFn: () =>
+        api
+          .put<Group>(`/group/${group.id}`, { name: groupNameText })
+          .then((res) => res.data),
+      onSuccess: () => {
+        queryClient.invalidateQueries([QueryKeys.FETCH_GROUP_INFO]);
+        toast.custom(<ToastMessage message="✅ Group Name Changed" />);
+      },
+      onError: () => {
+        setChangeGroupNameError(
+          "Error changing group name, check your internet connection.",
+        );
+      },
+    });
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  useEffect(() => {
+    if (!isFirstRender && groupNameText.trim() === "") {
+      setChangeGroupNameError("Group name cannot be empty.");
+    } else {
+      setIsFirstRender(false);
+      setChangeGroupNameError("");
+    }
+  }, [groupNameText]);
+
+  const { mutate: leaveGroup, isLoading: leaveGroupLoading } = useMutation({
+    mutationFn: () =>
+      api.put<Group>(`/group/${group.id}`, { remove_users: [user!.username] }),
+    onSuccess: () => {
+      queryClient.invalidateQueries([QueryKeys.FETCH_GROUP_INFO]);
+      toast.custom(<ToastMessage message="🚪 Left The Group" />);
+      navigate("/groups");
+    },
+    onError: () => {
+      setLeaveError("Error leaving group, check your internet connection.");
+    },
+  });
+  return (
+    <>
+      <Column className="mx-8 gap-6 pt-16">
+        <Row
```

Comment:
Super minor but rn the cursor doesn't change when you hover over the back button. Can you make it so it changes to the click button cursor?

Reply:
thanks for catching that!